codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/StructureGenerator.kt

```diff
@@ -113,6 +120,42 @@ class StructureGenerator(
         }
     }
 
+    private fun renderStructureImpl() {
+        if (members.isEmpty()) {
+            return
+        }
+        writer.rustBlock("impl $name") {
+            // Render field accessor methods
+            forEachMember { member, memberName, memberSymbol ->
+                // Let the ErrorGenerator render the error message accessor if this is an error struct
+                if (errorTrait == null || memberName != "message") {
+                    renderMemberDoc(member, memberSymbol)
+                    val memberType = memberSymbol.rustType()
+                    val returnType = if (memberType.isCopy()) {
+                        memberType
+                    } else if (memberType is RustType.Option && memberType.member.isDeref()) {
+                        memberType.asDeref()
+                    } else {
+                        memberType.asRef()
+                    }
+                    rustBlock("pub fn $memberName(&self) -> ${returnType.render()}") {
+                        if (memberType.isCopy()) {
+                            rust("self.$memberName")
+                        } else if (memberType is RustType.Option && memberType.member.isDeref()) {
+                            rust("self.$memberName.as_deref()")
+                        } else if (memberType is RustType.Option) {
+                            rust("self.$memberName.as_ref()")
+                        } else if (memberType.isDeref()) {
+                            rust("self.$memberName.deref()")
+                        } else {
+                            rust("&self.$memberName")
+                        }
```

Comment:
totally up to you, but for these long chains, sometimes a when block where the LHS and RHS can be on the same line are easier to read:
```kotlin
                        when {
                            memberType.isCopy() -> rust("self.$memberName")
                            // ...
                         }
```